src/js/src/views/HarvestCalendar.vue

```diff
@@ -0,0 +1,162 @@
+        <span class="pointer" :class="{active: view === 'all-years'}" @click="showAllYears">Days</span>
+      </p>
+      <transition name="fadeIn">
+        <year-month-graph v-if="view === 'year-month'" :harvest-data="harvestData" :show-year-details="showYearWeek" />
+      </transition>        
+      <transition name="fadeIn">
+        <week-graph v-if="view === 'year-week'"
+                    :year="year"
+                    :harvest-data="harvestData"
+                    :show-all="showYearMonth"
+                    :url="currentHarvestUrl"
+                    class="detailsContainer" />
+      </transition> 
+      <transition name="fadeIn">
+        <all-years-graph v-if="view === 'all-years'" :harvest-data="harvestData" class="detailsContainer" />
+      </transition>
+    </div>
+    <div v-if="!harvestData && noResults">
+      <p>No results.</p>
+    </div> 
+    <!-- TODO implement new spinner function when ready -->           
+    <div v-if="!harvestData && !noResults">
+      <div>
+        <p class="spinnerText">
+          Fetching harvests
+        </p>
+      </div>
+    </div>
+  </div>     
+</template>
+
+<style lang="scss">
+  @import '../assets/styles/harvestCalendar.scss'; 
+</style>
+
+<script>
+import { requestService } from '../services/RequestService'
+import {groupHarvestDatesByYearAndMonth} from '../components/harvestCalendar/tranformers/transformerMain'
+import {calculateLinearActivityLevel, calculateLogarithmicActivityLevel} from '../components/harvestCalendar/plugins/tranformationHelpers'
+import {toHumanDate} from '../components/harvestCalendar/util'
+import YearMonthGraph from '../components/harvestCalendar/YearMonthGraph'
+import AllYearsGraph from '../components/harvestCalendar/AllYearsGraph'
+import WeekGraph from '../components/harvestCalendar/WeekGraph'
+import Notifications from '../components/notifications/Notifications'
+import { mapActions } from 'vuex'
+import Vue from 'vue'
+import VTooltip from 'v-tooltip'
+
+Vue.use(VTooltip)
+
+
+export default {
+  name: 'HarvestCalendar',
+  components: {  
+    YearMonthGraph,
+    AllYearsGraph,
+    WeekGraph,
+    Notifications
+  },
+
+  filters: {
+  'human-date': function (date, showWeekday = false) {
+      return toHumanDate(date, showWeekday)
+    },
+   'formatted-number': function (value) {
+       if (!isNaN(value)) {
+        return value.toLocaleString()
+    }
+    return value
+   }
+  },
+  
+  data () {
+    return {
+            harvestData: null,
+            view: 'year-month',
+            year: null,
+            noResults: false,
+            currentHarvestUrl: ''
+        }
+  },
+
+  
+  
+  mounted () {
+    if (this.$route.query.url){
+       this.currentHarvestUrl = this.$route.query.url
+     }
+     requestService.getHarvestDates(encodeURIComponent(this.$route.query.url))
+        .then(data => {
+            if (data.dates === undefined || data.dates.length === 0) {
```

Comment:
Is this double check necessary? Im just curious, I don't know the returned data...

Reply:
No - I double checked (no pun intended :) )  only length check is needed. Good catch.